### PR TITLE
Add pre-suffle weight for new aiter MoE support.

### PR DESCRIPTION
**Pull request by sogalin**
## Motivation

This PR introduces a pre-shuffle step for MoE weights to improve runtime performance and memory access efficiency.

## Modifications

Added shuffle_weight from aiter.ops.shuffle to pre-shuffle w13_weight and w2_weight with a (16, 16) granularity matching aiter kernel tile size for better locality and performance.

## Accuracy Tests

Model: amd/DeepSeek-R1-MXFP4-Preview

python3 benchmark/gsm8k/bench_sglang.py --num-questions 2000 --parallel 2000 --port 8000
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1319/1319 [01:50<00:00, 11.96it/s]
Accuracy: 0.946
Invalid: 0.000
Latency: 110.434 s
Output throughput: 1180.643 token/s

## Benchmarking and Profiling

Machine: MI355 * 8 GPU
Docker Image: rocm/sgl-dev:v0.5.4.post3-rocm700-mi35x-20251106

Command:
SGLANG_USE_AITER=1 RCCL_MSCCL_ENABLE=0 SGLANG_INT4_WEIGHT=0 SGLANG_MOE_PADDING=1 SGLANG_USE_ROCM700A=1 SGLANG_SET_CPU_AFFINITY=1 SGLANG_ROCM_FUSED_DECODE_MLA=1 python3 -m sglang.launch_server --model-path /data2/deepseek-ai/DeepSeek-R1-MXFP4-Preview/ --tensor-parallel-size 8 --trust-remote-code --chunked-prefill-size 131072 --host 0.0.0.0 --port 8000 --log-requests  --disable-radix-cache --mem-fraction-static 0.5 --speculative-algorithm EAGLE --speculative-num-steps 3 --speculative-eagle-topk 1 --speculative-num-draft-tokens 4

Prefill 1 8 16
Wo MoE Update 0.0109 0.6524 1.2924
With MoE Update 0.1071 0.6505 1.2965

Decode 1 8 16
Wo MoE Update 0.0109 0.0123 0.0139
With MoE Update 0.0103 0.0118 0.0133
## Checklist

- [X] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/layers/quantization/mxfp4.py [hunk 1]
```diff
@@ -71,12 +72,14 @@
     )
 
 _is_hip = is_hip()
+_is_shuffle_moe = get_bool_env_var("AITER_MXFP4_MOE_SF") and _is_hip
```
Comment (HaiShaw) [inline python/sglang/srt/layers/quantization/mxfp4.py]: suggest change to `_is_shuffle_moe_mxfp4`

File: python/sglang/srt/layers/quantization/quark/quark_moe.py [hunk 2]
```diff
@@ -24,6 +25,7 @@
 logger = logging.getLogger(__name__)
 
 _is_hip = is_hip()
+_is_shuffle_moe = get_bool_env_var("AITER_MXFP4_MOE_SF") and _is_hip
```
Comment (HaiShaw) [inline python/sglang/srt/layers/quantization/quark/quark_moe.py]: change to `_is_shuffle_moe_mxfp4`

**Review by HaiShaw — CHANGES_REQUESTED** [review]

**Review by HaiShaw — APPROVED** [review]

File: python/sglang/srt/layers/quantization/quark/quark_moe.py [hunk 3]
```diff
@@ -167,6 +169,15 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
         # layer.w2_weight_scale = torch.nn.Parameter(w2_weight_scale, requires_grad=False)
         layer.w2_weight_scale.data = w2_weight_scale.view(s0, s1, -1)
 
+        # Pre-shuffle weight
+        if _is_shuffle_moe_mxfp4:
```
Comment (BowenBao) [inline python/sglang/srt/layers/quantization/quark/quark_moe.py:173]: @sogalin for my understanding, why would this not need any changes on the kernel call code down in the `apply` method?

@HaiShaw Would it make sense to keep shuffling as default since it has better perf?

Comment (HaiShaw) [reply python/sglang/srt/layers/quantization/quark/quark_moe.py:173]: @BowenBao it is set in docker ENV
